src/Microsoft.ML.Data/DataLoadSave/Text/TextLoader.cs

```diff
@@ -91,10 +105,18 @@ public Column(string name, DataKind? type, Range[] source, KeyCount keyCount = n
             public string Name;
 
             /// <summary>
-            /// <see cref="DataKind"/> of the items in the column. If <see langword="null"/> defaults to a float.
+            /// <see cref="InternalDataKind"/> of the items in the column. If <see langword="null"/> defaults to a float.
+            /// Although <see cref="InternalDataKind"/> is internal, <see cref="Type"/>'s information can be publically accessed by <see cref="ItemType"/>.
             /// </summary>
             [Argument(ArgumentType.AtMostOnce, HelpText = "Type of the items in the column")]
-            public DataKind? Type;
+            [BestFriend]
+            internal InternalDataKind? Type;
```

Comment:
>InternalDataKind? Type; [](start = 21, length = 23)

Would be great if you make it non-nullable field.
It doesn't have to.

Reply:
Can we break entry-point BC? I can remove nullable if your answer is yes.

---
In reply to: [259543165](https://github.com/dotnet/machinelearning/pull/2661#discussion_r259543165) [](ancestors = 259543165)

Reply:
Please do. Just make it:
`internal InternalDataKind Type = InternalDataKind.R4`


---
In reply to: [259549751](https://github.com/dotnet/machinelearning/pull/2661#discussion_r259549751) [](ancestors = 259549751,259543165)

Reply:
Why `InternalDataKind Type = default` default?

---
In reply to: [259551578](https://github.com/dotnet/machinelearning/pull/2661#discussion_r259551578) [](ancestors = 259551578,259549751,259543165)

Reply:
The reason is described (indirectly) in Iteration 9. TextLoader doesn't store default settings so that some baseline files need to be updated.

---
In reply to: [259555563](https://github.com/dotnet/machinelearning/pull/2661#discussion_r259555563) [](ancestors = 259555563,259551578,259549751,259543165)